packages/flutter/lib/src/material/dropdown.dart

```diff
@@ -768,19 +788,29 @@ class _DropdownButtonState<T> extends State<DropdownButton<T>> with WidgetsBindi
     return math.max(_textStyle.fontSize, math.max(widget.iconSize, _kDenseButtonHeight));
   }
 
-  Color get _downArrowColor {
+  Color get _iconColor {
     // These colors are not defined in the Material Design spec.
     if (_enabled) {
-      if (Theme.of(context).brightness == Brightness.light) {
-        return Colors.grey.shade700;
-      } else {
-        return Colors.white70;
+      if (widget.iconEnabledColor != null) {
+        return widget.iconEnabledColor;
+      }
+
+      switch(Theme.of(context).brightness) {
+        case Brightness.light:
+          return Colors.grey.shade700;
+        default:
+          return Colors.white70;
       }
     } else {
-      if (Theme.of(context).brightness == Brightness.light) {
-        return Colors.grey.shade400;
-      } else {
-        return Colors.white10;
+      if (widget.iconDisabledColor != null) {
+        return widget.iconDisabledColor;
+      }
+
+      switch(Theme.of(context).brightness) {
+        case Brightness.light:
+          return Colors.grey.shade400;
+        default:
```

Comment:
We use [switch for enums](https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#avoid-using-if-chains-or--or--with-enum-values) so that an error generated if the enum is extended. So `Brightness.dark` instead of default.

Reply:
without the `default` case, Flutter returns the following warning:
```
warning • This function has a return type of 'Color', but doesn't end with a
       return statement • lib/src/material/dropdown.dart:791:3 • missing_return

1 issue found. (ran in 9.4s)
```

How should I proceed? The alternatives I can think of are keeping the `default` case or having a `return null` or some default Color value at the end

Reply:
Resolved offline, appropriate changes have been made